pkg/meta/kvclient/client.go

```diff
@@ -1,30 +1,54 @@
 package kvclient
 
 import (
+	"context"
+	"time"
+
+	"github.com/hanfei1991/microcosm/pkg/meta/extension"
 	"github.com/hanfei1991/microcosm/pkg/meta/kvclient/etcdkv"
 	"github.com/hanfei1991/microcosm/pkg/meta/metaclient"
 	"github.com/hanfei1991/microcosm/pkg/meta/namespace"
+	"github.com/hanfei1991/microcosm/pkg/tenant"
 )
 
 // etcdKVClient is the implement of kv interface based on etcd
 // Support namespace isolation and all kv ability
-// etcdImpl -> kvPrefix+etcdClientCloser -> etcdKVClient
+// etcdImpl -> kvPrefix+Closer -> etcdKVClient
 type etcdKVClient struct {
 	metaclient.Closer
 	metaclient.KV
 	tenantID string
 }
 
-func NewEtcdKVClient(config *metaclient.Config, tenantID string) (metaclient.KVClient, error) {
-	impl, err := etcdkv.NewEtcdImpl(config)
-	if err != nil {
-		return nil, err
-	}
-
-	pfKV := namespace.NewPrefixKV(impl, namespace.MakeNamespacePrefix(tenantID))
+// NewPrefixKVClient return a kvclient with namespace
+func NewPrefixKVClient(cli extension.KVClientEx, tenantID string) metaclient.KVClient {
+	pfKV := namespace.NewPrefixKV(cli, namespace.MakeNamespacePrefix(tenantID))
 	return &etcdKVClient{
-		Closer:   etcdkv.NewEtcdClientCloser(impl),
+		Closer:   cli,
 		KV:       pfKV,
 		tenantID: tenantID,
-	}, nil
+	}
+}
+
+// NewKVClient return a kvclient without namespace for inner use
+func NewKVClient(conf *metaclient.StoreConfigParams) (extension.KVClientEx, error) {
+	return etcdkv.NewEtcdImpl(conf)
+}
+
+func CheckAccessForMetaStore(conf *metaclient.StoreConfigParams) error {
```

Comment:
Is this a connectivity test, or will add privilege check later, please add some comments.